receiver/receivercreator/config.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package receivercreator
+
+import (
+	otelconfig "github.com/open-telemetry/opentelemetry-collector/config"
+	"github.com/open-telemetry/opentelemetry-collector/config/configmodels"
+)
+
+// subreceiverConfig is the configuration of a single subreceiver configured inside
+// receiver_creator.
+type subreceiverConfig struct {
+	// Rule is the discovery rule that when matched will create a receiver instance
+	// based on subreceiverConfig.
+	Rule string `mapstructure:"rule"`
+
+	// receiverType is set based on the configured receiver name.
+	receiverType string
+	// config is a map containing the default set of values used when instantiating
+	// the receiver at runtime.
```

Comment:
It would be nice to describe what it the key of the map. Even better would be to define this `map[string]interface{}` as a type alias since it is also used elsewhere (e.g. as function parameters).

Reply:
I've updated the comment, see if it helps. It's hard to describe the keys since the whole map is arbitrarily defined by the user but I gave it a shot.